### PR TITLE
♿  Chore/improve logging

### LINKED ISSUE #104: Improve logging
Improve logging:
- have the same logging format everywhere
- have info logs when jobs, apps, pipelines....are successfully created
- in jobs with callbacks, remove logs every x seconds an replace them wuith a status in the console only

### PR DESCRIPTION
**Pull request by tquiviger**
Fixes #104

### CODE REVIEW

File: saagieapi/apps/apps.py [hunk 1]
```diff
@@ -112,7 +118,7 @@ def create_from_scratch(
         check_format_exposed_port = self.check_exposed_ports(exposed_ports)
         if not check_format_exposed_port:
             raise ValueError(
-                "The parameter 'exposed_ports' should be a list of dict. Each dict should contains the key 'port'."
+                "❌ The parameter 'exposed_ports' should be a list of dict. Each dict should contains the key 'port'."
                 "All accept key of each dict is: '{list_exposed_port_field}'"
```
Comment (Qiwei08) [inline saagieapi/apps/apps.py:122]: Should add fstring for l116, like:
` f"All accept key of each dict is: '{list_exposed_port_field}'"`

File: saagieapi/apps/apps.py [hunk 2]
```diff
@@ -247,7 +254,7 @@ def create_from_catalog(
...
```
Comment (Qiwei08) [inline saagieapi/apps/apps.py:258]: Should add fstring for l258, like:
` f"All accept key of each dict is: '{list_exposed_port_field}'"`

File: saagieapi/apps/apps.py [hunk 3]
```diff
@@ -18,27 +19,32 @@ def list_for_project(self, project_id: str) -> Dict:
         ----------
         project_id : str
             UUID of your project (see README on how to find it)
+
```
Comment (Qiwei08) [inline saagieapi/apps/apps.py:22]: Add `pprint_result: Optional[bool] = True` as parameter ?

File: saagieapi/docker_credentials/docker_credentials.py [hunk 4]
```diff
@@ -9,20 +10,24 @@ class DockerCredentials:
...
         dict
 
         """
         params = {"projectId": project_id}
-        return self.saagie_api.client.execute(gql(GQL_GET_ALL_DOCKER_CREDENTIALS), variable_values=params)
+        return self.saagie_api.client.execute(
+            query=gql(GQL_GET_ALL_DOCKER_CREDENTIALS), variable_values=params, pprint_result=pprint_result
+        )
 
     def get_info(self, project_id: str, credential_id: str) -> Dict:
```
Comment (Qiwei08) [inline saagieapi/docker_credentials/docker_credentials.py:32]: Add `pprint_result` as parameter ?

File: saagieapi/env_vars/env_vars.py [hunk 5]
```diff
@@ -82,10 +87,10 @@ def update_global(
             When the variable doesn't already exist
         """
 
-        existing_env_var = self.list_globals()["globalEnvironmentVariables"]
+        existing_env_var = self.list_globals(pprint_result=False)["globalEnvironmentVariables"]
 
         if name not in [env_var["name"] for env_var in existing_env_var]:
-            raise ValueError("Environment variable does not exists")
+            raise ValueError("❌ Environment variable does not exists")
```
Comment (Qiwei08) [inline saagieapi/env_vars/env_vars.py:93]: Maybe it's better to specify the name that doesn't exist as environment variable like:

`f"❌ Environment variable {name} does not exists"`

File: saagieapi/env_vars/env_vars.py [hunk 6]
```diff
@@ -257,10 +269,10 @@ def update_for_project(
             When the variable doesn't already exist
         """
 
-        existing_env_var = self.list_for_project(project_id)["projectEnvironmentVariables"]
+        existing_env_var = self.list_for_project(project_id, pprint_result=False)["projectEnvironmentVariables"]
 
         if name not in [env_var["name"] for env_var in existing_env_var]:
-            raise ValueError("Environment variable does not exists")
+            raise ValueError("❌ Environment variable does not exists")
```
Comment (Qiwei08) [inline saagieapi/env_vars/env_vars.py:275]: Maybe it's better to specify the name that doesn't exist as environment variable like:

`f"❌ Environment variable {name} does not exists"`

File: saagieapi/jobs/jobs.py [hunk 7]
```diff
@@ -33,8 +34,10 @@ def list_for_project(self, project_id: str, instances_limit: int = -1) -> Dict:
         params = {"projectId": project_id}
         if instances_limit != -1:
             params["instancesLimit"] = instances_limit
-        query = gql(GQL_LIST_JOBS_FOR_PROJECT)
-        return self.saagie_api.client.execute(query, variable_values=params)
+
+        return self.saagie_api.client.execute(
+            query=gql(GQL_LIST_JOBS_FOR_PROJECT), variable_values=params, pprint_result=True
```
Comment (Qiwei08) [inline saagieapi/jobs/jobs.py:39]: Add `pprint_result` as parameter ?

File: saagieapi/pipelines/pipelines.py [hunk 8]
```diff
@@ -32,8 +33,9 @@ def list_for_project(self, project_id: str, instances_limit: int = -1) -> Dict:
         params = {"projectId": project_id}
         if instances_limit != -1:
             params["instancesLimit"] = instances_limit
-        query = gql(GQL_LIST_PIPELINES_FOR_PROJECT)
-        return self.saagie_api.client.execute(query, variable_values=params)
+        return self.saagie_api.client.execute(
+            query=gql(GQL_LIST_PIPELINES_FOR_PROJECT), variable_values=params, pprint_result=True
```
Comment (Qiwei08) [inline saagieapi/pipelines/pipelines.py:37]: `pprint_result` as parameter ?

File: saagieapi/projects/projects.py [hunk 9]
```diff
@@ -53,8 +54,7 @@ def list(self) -> Dict:
         dict
             Dict of projects information
         """
-        query = gql(GQL_LIST_PROJECTS)
-        return self.saagie_api.client.execute(query)
+        return self.saagie_api.client.execute(query=gql(GQL_LIST_PROJECTS), pprint_result=True)
```
Comment (Qiwei08) [inline saagieapi/projects/projects.py:57]: `pprint_result` as parameter?

File: saagieapi/projects/projects.py [hunk 10]
```diff
@@ -72,7 +72,7 @@ def get_id(self, project_name: str) -> Dict:
         if project:
             project_id = project[0]["id"]
             return project_id
-        raise NameError(f"Project {project_name} does not exist or you don't have permission to see it.")
+        raise NameError(f"❌ Project {project_name} does not exist or you don't have permission to see it.")
 
     def get_info(self, project_id: str) -> Dict:
```
Comment (Qiwei08) [inline saagieapi/projects/projects.py:77]: `pprint_result` as parameter ?

File: saagieapi/apps/apps.py (hunk 2 above)
Comment (tquiviger) [reply saagieapi/apps/apps.py]: done

File: saagieapi/apps/apps.py (hunk 3 above)
Comment (tquiviger) [reply saagieapi/apps/apps.py:24]: not needed as this function is not called by other functions

File: saagieapi/docker_credentials/docker_credentials.py (hunk 4 above)
Comment (tquiviger) [reply saagieapi/docker_credentials/docker_credentials.py:32]: not needed as this function is not called by other functions

File: saagieapi/env_vars/env_vars.py (hunk 5 above)
Comment (tquiviger) [reply saagieapi/env_vars/env_vars.py]: done

File: saagieapi/env_vars/env_vars.py (hunk 6 above)
Comment (tquiviger) [reply saagieapi/env_vars/env_vars.py]: done

File: saagieapi/jobs/jobs.py (hunk 7 above)
Comment (tquiviger) [reply saagieapi/jobs/jobs.py:39]: not needed as this function is not called by other functions

File: saagieapi/pipelines/pipelines.py (hunk 8 above)
Comment (tquiviger) [reply saagieapi/pipelines/pipelines.py:37]: not needed as this function is not called by other functions

File: saagieapi/projects/projects.py (hunk 9 above)
Comment (tquiviger) [reply saagieapi/projects/projects.py:57]: not needed as this function is not called by other functions

File: saagieapi/projects/projects.py (hunk 10 above)
Comment (tquiviger) [reply saagieapi/projects/projects.py:77]: not needed as this function is not called by other functions

**Review by Qiwei08 — APPROVED** [review]

File: saagieapi/jobs/jobs.py [hunk 11]
```diff
@@ -522,18 +556,24 @@ def run_with_callback(self, job_id: str, freq: int = 10, timeout: int = -1) -> D
...
-                raise TimeoutError("Last state known : " + state)
-            time.sleep(freq)
-            sec += freq
-            job_instance_info = self.get_instance(job_instance_id)
-            state = job_instance_info.get("jobInstance").get("status")
-            logging.info(f"Job id {job_id} with instance {job_instance_id} is currently : " + state)
+            with console.status(f"Job is currently {state}", refresh_per_second=100):
+                to = False if timeout == -1 else sec >= timeout
+                if to:
+                    raise TimeoutError("❌ Last state known : " + state)
```
Comment (guillaumeprevost76) [inline saagieapi/jobs/jobs.py:568]: Should add fstring for l568, like:
raise TimeoutError(f"❌ Last state known : {state}")

File: saagieapi/pipelines/pipelines.py [hunk 12]
```diff
@@ -391,20 +411,33 @@ def run_with_callback(self, pipeline_id: str, freq: int = 10, timeout: int = -1)
...
-                raise TimeoutError("Last state known : " + state)
-            time.sleep(freq)
-            sec += freq
-            pipeline_instance_info = self.get_instance(pipeline_instance_id)
-            state = pipeline_instance_info.get("pipelineInstance").get("status")
-            logging.info("Current state : %s", state)
+            with console.status(f"Job is currently {state}", refresh_per_second=100):
+                to = False if timeout == -1 else sec >= timeout
+                if to:
+                    raise TimeoutError("Last state known : " + state)
```
Comment (guillaumeprevost76) [inline saagieapi/pipelines/pipelines.py:426]: Should add fstring and emoji for l568, like:
raise TimeoutError(f"x Last state known : {state}")

**Review by guillaumeprevost76 — APPROVED** [review]

**Pull request merged**